packages/oauth-twitter/src/index.ts

```diff
@@ -0,0 +1,52 @@
+import * as oauth from 'oauth';
+
+export interface AccountsOauthTwitterOptions {
+  key: string;
+  secret: string;
+}
+
+class AccountsOauthTwitter {
+  private options: AccountsOauthTwitterOptions;
+
+  constructor(options: AccountsOauthTwitterOptions) {
+    this.options = options;
+  }
+
+  public authenticate(params) {
+    const oa = new oauth.OAuth(
+      'https://twitter.com/oauth/request_token',
+      'https://twitter.com/oauth/access_token',
+      this.options.key,
+      this.options.secret,
+      '1.0A',
+      null,
+      'HMAC-SHA1',
+    );
+    return new Promise((resolve, reject) => {
+      oa.get(
+        'https://api.twitter.com/1.1/account/verify_credentials.json?include_email=true',
+        params.access_token,
+        params.access_token_secret,
+        (err, data) => {
+          if (err) {
+            reject(err);
+          } else {
+            data = JSON.parse(data);
+            const user = {
+              id: data.id_str,
+              screenName: data.screen_name,
```

Comment:
What are your thoughts on returning the entirety of `data` response and allowing the developer to decide if they want to consume the additional data (part of `@accounts/server`). The only fields which are required to be returned here are some sort of unique identifier and the tokens correct?

Reply:
Yeah you are right, I just took the same implementation as meteor.
Maybe we could allow a custom user function which indicate which fields should be saved?